### PR TITLE
Add a common package for backend jwt generation.

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
This PR will add a common package for backend jwt generation.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway.common/src/main/java/org/wso2/carbon/apimgt/gateway/common/jwtgenerator/ClaimsRetriever.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * This interface encapsulates a user claims retriever.
+ * The retrieved claims are encoded to the JWT during subscriber validation
+ * in the order defined by the SortedMap.
+ * Anyone trying to add custom user properties to the JWT should implement this interface
+ * and mention the fully qualified class name in api-manager.xml ->
+ * JWTConfiguration -> ClaimsRetrieverImplClass
+ */
+public interface ClaimsRetriever {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway.common/src/main/java/org/wso2/carbon/apimgt/gateway/common/jwtgenerator/ClaimsRetriever.java:15]: keep Claim Retriever in same place and bing the dialect uri

File: components/apimgt/org.wso2.carbon.apimgt.gateway/pom.xml [hunk 2]
```diff
@@ -315,6 +315,10 @@
             <groupId>org.wso2.carbon.apimgt</groupId>
             <artifactId>org.wso2.carbon.apimgt.rest.api.util</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.am.analytics.publisher</groupId>
+            <artifactId>org.wso2.am.analytics.publisher.client</artifactId>
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/pom.xml:320]: does it intentional

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 3]
```diff
@@ -177,9 +197,13 @@ public AuthenticationContext authenticate(SignedJWTInfo signedJWTInfo, MessageCo
                 log.debug("JWT authentication successful.");
                 String endUserToken = null;
                 if (jwtGenerationEnabled) {
+                    log.debug("SHALKI:Inside jwtGenerationEnabled>>>>>>>>>>");
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java:200]: remove these lines

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/generator/APIMgtGatewayUrlSafeJWTGeneratorImpl.java [hunk 4]
```diff
@@ -19,12 +19,13 @@
...
 import org.apache.commons.codec.binary.Base64;
-import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.gateway.common.exception.JWTGeneratorException;
+import org.wso2.carbon.apimgt.gateway.common.jwtgenerator.APIMgtGatewayJWTGeneratorImpl;
 
 public class APIMgtGatewayUrlSafeJWTGeneratorImpl extends APIMgtGatewayJWTGeneratorImpl {
 
     @Override
-    public String encode(byte[] stringToBeEncoded) throws APIManagementException {
+    public String encode(byte[] stringToBeEncoded) throws JWTGeneratorException {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/generator/APIMgtGatewayUrlSafeJWTGeneratorImpl.java:28]: move this class to common package

File: components/apimgt/org.wso2.carbon.apimgt.tracing/pom.xml [hunk 5]
```diff
@@ -69,6 +69,11 @@
             <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>io.opentracing</groupId>
+            <artifactId>opentracing-api</artifactId>
+            <version>0.31.0</version>
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.tracing/pom.xml:75]: is this change intentionally to this pr.

**Review by tharindu1st — CHANGES_REQUESTED** [review]

**Comment by tharindu1st** [comment]
add license header to all missing files

File: components/apimgt/org.wso2.carbon.apimgt.tracing/pom.xml (hunk 5 above)
Comment (ShalkiWenushika) [reply components/apimgt/org.wso2.carbon.apimgt.tracing/pom.xml]: yes. Without this dependancy, I get a compilation error

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**